identifier/src/containers/Login/Login.jsx

```diff
@@ -83,6 +84,11 @@ function Login(props) {
       history.replace(`/chooseaccount${history.location.search}${history.location.hash}`);
       return;
     }
+
+    if (query && query.login_hint){
+      const sanitizedHint = DOMPurify.sanitize(query.login_hint);
```

Comment:
These can be directly coming from the query parameter of the identifier route. 

For example, it can be used like this :

![image](https://github.com/user-attachments/assets/28c6b19b-09e5-4990-b6b5-d9ee3cbd06bc)


and this is added as a parameter for the url

https://host/signin/v1/identifier?client_id=e4a-manage-http%3A%2F%2Fs0ap%3A3099%2Fmanage&code_challenge=tJSII2fOZY6d6FKmqKZ5MXWDkHAfB-fP8w43bqwJFVk&code_challenge_method=S256&flow=oidc&login_hint=admin%40exchange4all.local&prompt=select_account&redirect_uri=http%3A%2F%2Fs0ap%3A3099%2Fmanage%2F%23%2Foidc%2Fauthorize-callback%3F&response_mode=fragment&response_type=code&scope=openid+profile+email+E4A.Manage.Default&state=4feb15d37b3e42f28dcaaeb94f806bbf

Now the login_hint from the parameter is used for the username. Now since this can be modified to have scripts or any sort of attack string ( especially XSS attacks ). I added Dompurify to sanitize the string directly before adding it as a value.

Reply:
Ok, validate the newly added query parameter like so:

1. Use https://www.npmjs.com/package/validator
2. Validate with the `isEmail` function
3. If `isEmail` is not valid, validate with `isEmail` again but extend the string with `@example.com` (for validation only)
4. If its not valid after going through these steps, ignore the query string value

Reply:
Hi @longsleep , updated the validator as per mentioned. So now it behaves something like this:

best case :

![image](https://github.com/user-attachments/assets/9e947335-0eda-4840-a36d-4092ab077a49)

on this shows like 

![image](https://github.com/user-attachments/assets/13373c4f-dde2-4acc-8490-843198dbb7d7)


moderate case, no email address but user name provided:
![image](https://github.com/user-attachments/assets/5f55922f-8fc5-4d92-a36a-3b5a4b8c6d6f)

shows like this 

![image](https://github.com/user-attachments/assets/c5ebc871-91e9-48b0-904e-b52e31f071b9)

worst case:

in inputs like this 

![image](https://github.com/user-attachments/assets/bd85dc72-5253-4a15-8e33-953c844c1067)

it will show something like this

![image](https://github.com/user-attachments/assets/cf9ce944-00b3-43b9-9e8e-efa7b710296d)